conferences/ticketing_storage.go

```diff
@@ -0,0 +1,605 @@
+	WHERE attendees_to_slot_claims.id = $1`,
+		results.ID)
+	if err != nil && err != sql.ErrNoRows {
+		return nil, fmt.Errorf("querying claims for attendee: %w", err)
+	}
+	for rows.Next() {
+		claim := SlotClaim{}
+		err := rows.Scan(&claim.ID, &claim.TicketID, &claim.Redeemed)
+		if err != nil {
+			return nil, fmt.Errorf("scanning slot_claims for attendee: %w", err)
+		}
+		claims = append(claims, claim)
+	}
+
+	results.Claims = claims
+	return &results, nil
+}
+
+// createConferenceSlot saves a slot in the database.
+func createConferenceSlot(ctx context.Context, tx *sqldb.Tx, cslot *ConferenceSlot, conferenceID int64) (*ConferenceSlot, error) {
+	var sqlSentence = `INSERT INTO conference_slot (conference_id, name, descripcion, cost, capacity, start_date, end_date, purchaseable_form, purchaseable_until, available_to_public)
+	VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10) 
+	RETURNING conference_id, name, descripcion, cost, capacity, start_date, end_date, purchaseable_form, purchaseable_until, available_to_public`
+	var args = []interface{}{
+		conferenceID,
+		cslot.Name,
+		cslot.Description,
+		cslot.Cost,
+		cslot.Capacity,
+		cslot.StartDate,
+		cslot.EndDate,
+		cslot.PurchaseableFrom,
+		cslot.PurchaseableUntil,
+		cslot.AvailableToPublic,
+	}
+	/*if e.DependsOn != nil {
+		sqlSentence = `INSERT INTO conference_slot (conference_id, name, descripcion, cost, capacity, start_date, end_date, purchaseable_form, purchaseable_until, available_to_public, depends_on_id)
+		VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11)
+		RETURNING id, name, descripcion, cost, capacity, start_date, end_date, purchaseable_form, purchaseable_until, available_to_public`
+		args = append(args, e.DependsOn.ID)
+	}*/
+	row := queryRow(ctx, tx, sqlSentence, args...)
+	results := ConferenceSlot{}
+	err := row.Scan(&results.ID,
+		&results.Name,
+		&results.Description,
+		&results.Cost,
+		&results.Capacity,
+		&results.StartDate,
+		&results.EndDate,
+		&results.PurchaseableFrom,
+		&results.PurchaseableUntil,
+		&results.AvailableToPublic)
+	if err != nil {
+		return nil, fmt.Errorf("creating new attendee: %w", err)
+	}
+
+	// results.DependsOn = e.DependsOn
+
+	return &results, nil
+}
+
+func readConferenceSlotByID(ctx context.Context, tx *sqldb.Tx, id uint64, loadDeps bool) (*ConferenceSlot, error) {
+	results := ConferenceSlot{}
+	row := queryRow(ctx, tx,
+		`SELECT (id, name, descripcion, cost, capacity, start_date, end_date, purchaseable_form, purchaseable_until, available_to_public, conference_id, depends_on_id)
+	FROM conference_slot 
+	WHERE id = $1`, id)
+
+	var dependsOnID uint64
+	var conferenceID uint64
+
+	err := row.Scan(&results.ID,
+		&results.Name,
+		&results.Description,
+		&results.Cost,
+		&results.Capacity,
+		&results.StartDate,
+		&results.EndDate,
+		&results.PurchaseableFrom,
+		&results.PurchaseableUntil,
+		&results.AvailableToPublic,
+		&conferenceID,
+		&dependsOnID)
+	if err == sql.ErrNoRows {
+		return nil, nil
```

Comment:
Shouldn't this be an error if you look up a conference by ID and there isn't one?

Reply:
No, we do not guarantee one is there, `nil` is an acceptable response for "we don't have what you are looking for"

Reply:
Shouldn't we be returning a 404 in such a case? We could handle that here or as an error that we translate to 404.